src/app/services/tabs.service.ts

```diff
@@ -0,0 +1,90 @@
+import { Injectable } from '@angular/core';
+import {MatSnackBar} from '@angular/material';
+
+// Unique ID function
+const uuid = require('uuid/v1');
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TabsService {
+  // tslint:disable-next-line:max-line-length
+  // tabs = [{id: 1, title: 'To Do', cards: [{id: 1, cardTitle: 'Change Title', cardDesc: 'As a User I want the possibility to change the title'}] } , {id: 2, title: 'Doing', cards: [] }];
+  private newCard: { id: any, cardDesc: any; cardTitle: any };
+  private newTab: { id: any; title: any; cards: any[] };
+  private tabs = [];
+  // tslint:disable-next-line:variable-name
+  constructor(private _snackBar: MatSnackBar) {
+  }
+
+  getTabs() {
+    return this.tabs;
+  }
+
+// Add a tab to an Array
+  addTab(tabTitle) {
+  console.log(` tab title: ${tabTitle}`);
+  if (tabTitle !== '' || tabTitle === undefined) {
+      this.newTab = {
+        id: uuid(),
+        title: tabTitle,
+        cards: []
+      };
+      console.table(this.newTab);
+      // @ts-ignore
+      this.tabs.push(this.newTab);
+      this.openSnackBar(`The tab ${tabTitle} was created successfully`, 'OK');
+    } else {
+      alert(`The name couldn't be empty`);
+    }
+  }
+// Add a card to a tab
+  addCard(card, tabId) {
+    this.newCard = {
+      id: uuid(),
```

Comment:
it's better to have a model class for card

Reply:
Ok, Thanks!